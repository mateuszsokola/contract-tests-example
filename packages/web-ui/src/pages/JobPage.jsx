import React from 'react';
import Moment from 'react-moment';
import { Link } from 'react-router-dom';

import fetchJob from '../services/fetchJob.js';
import NotFound from '../components/NotFound';
import Loading from '../components/Loading';
import Salary from '../components/Salary';

class JobPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: false,
            error: null,
            item: null
        }
    }

    componentWillMount() {
        const { id, slug } = this.props.match.params;
        this.setState({ loading: true, error: null });

        fetchJob({ id, slug })
            .then(item => {
                this.setState({ loading: false, error: null, item });
            })
            .catch(error => {
                this.setState({ loading: false, error: error.message });
            })
    }

    render() {
        const { loading, error, item } = this.state;

        if (loading) {
            return <Loading />;
        }

        if (error) {
            return <NotFound />;
        }

        const dateToFormat = (new Date(item.createdAt));
        
        return (
            <div>
                <h1>{item.title}</h1>
                <span className='location'>{item.city}</span>
                <Salary from={item.salaryFrom} to={item.salaryTo} />
                <Moment fromNow={true} date={dateToFormat.toUTCString()}/>
            </div>
        )
    }
}

export default JobPage;