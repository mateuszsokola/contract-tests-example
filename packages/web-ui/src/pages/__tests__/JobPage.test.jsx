import React from 'react';
import { mount } from 'enzyme';

import JobPage from '../JobPage';

jest.mock('react-router-dom')

const EXPECTED_BODY = {
    id: 1,
    slug: 'senior-front-end-developer',
    title: 'Senior Front End Developer',
    createdAt: 'Sat Feb 10 2018 18:12:40 GMT+0100 (CET)',
    salaryFrom: 4340000,
    salaryTo: 5880000,
    city: 'Vienna'
};

describe('<JobPage />', () => {
    it('displays a job posting', () => {
        const match = {
            params: {
                id: 1,
                slug: 'senior-front-end-developer'
            }
        };
        const wrapper = mount(<JobPage match={match}/>);

        wrapper.setState({
            loading: false,
            error: null,
            item: EXPECTED_BODY
        })

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('displays a not found', () => {
        const match = {
            params: {
                id: 2,
                slug: 'non-existent-position'
            }
        };
        const wrapper = mount(<JobPage match={match}/>);

        wrapper.setState({
            loading: false,
            error: new Error('Not Found'),
            item: null
        })

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('displays a loading info', () => {
        const match = {
            params: {
                id: 1,
                slug: 'senior-front-end-developer'
            }
        };
        const wrapper = mount(<JobPage match={match}/>);

        wrapper.setState({
            loading: true,
            error: null,
            item: null
        })

        expect(wrapper.html()).toMatchSnapshot();
    });
})