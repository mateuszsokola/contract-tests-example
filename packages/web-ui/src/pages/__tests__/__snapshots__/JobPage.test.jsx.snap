// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobPage /> displays a job posting 1`] = `"<div><h1>Senior Front End Developer</h1><span class=\\"location\\">Vienna</span><span>43.400,00 – 58.800,00 EUR</span><time datetime=\\"1518282760000\\">a month ago</time></div>"`;

exports[`<JobPage /> displays a loading info 1`] = `"<span>Loading…</span>"`;

exports[`<JobPage /> displays a not found 1`] = `"<div><h1>Not Found</h1><span>Proceed to <a href=\\"/\\">job listing</a></span></div>"`;
